compiler.cpp

```diff
@@ -118,7 +118,7 @@ Function *Compiler::FunctionAST(const ast::Func &func) {
   for (auto &Arg : TheFunction->args())
     NamedValues[std::string(Arg.getName())] = &Arg;
 
-  if (Value *RetVal = OutExprAST(func.getBody())) {
+  if ([[maybe_unused]] Value *RetVal = OutExprAST(func.getBody())) {
```

Comment:
Why was [[maybe_unused]] added?

Reply:
`RetVal` isn't being used. We can see this when using clang 16. Since we have `-Wall -Werror` this will result in an `unused variable` warning.